pkg/eventprocessor/impl/executor/impl/executor_test.go

```diff
@@ -0,0 +1,265 @@
+package impl
+
+import (
+	"context"
+	"database/sql"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/stretchr/testify/require"
+	"github.com/textileio/go-tableland/internal/tableland"
+	"github.com/textileio/go-tableland/pkg/eventprocessor"
+	"github.com/textileio/go-tableland/pkg/eventprocessor/eventfeed"
+	"github.com/textileio/go-tableland/pkg/parsing"
+	parserimpl "github.com/textileio/go-tableland/pkg/parsing/impl"
+	"github.com/textileio/go-tableland/pkg/sqlstore/impl/system"
+	"github.com/textileio/go-tableland/pkg/tables/impl/ethereum"
+	"github.com/textileio/go-tableland/tests"
+)
+
+var chainID = 1337
+
+func TestReceiptExists(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	ex, _ := newExecutorWithTable(t, 0)
+
+	txnHash := "0x0000000000000000000000000000000000000000000000000000000000001234"
+
+	bs, err := ex.NewBlockScope(ctx, 0)
+	require.NoError(t, err)
+	ok, err := bs.TxnReceiptExists(ctx, common.HexToHash(txnHash))
+	require.NoError(t, err)
+	require.False(t, ok)
+	require.NoError(t, bs.Commit())
+	require.NoError(t, bs.Close())
+
+	bs, err = ex.NewBlockScope(ctx, 0)
+	require.NoError(t, err)
+	err = bs.SaveTxnReceipts(ctx, []eventprocessor.Receipt{
+		{
+			ChainID:     tableland.ChainID(chainID),
+			BlockNumber: 100,
+			TxnHash:     txnHash,
+		},
+	})
+	require.NoError(t, err)
+	require.NoError(t, bs.Commit())
+	require.NoError(t, bs.Close())
+
+	bs, err = ex.NewBlockScope(ctx, 0)
+	require.NoError(t, err)
+	ok, err = bs.TxnReceiptExists(ctx, common.HexToHash(txnHash))
+	require.NoError(t, err)
+	require.True(t, ok)
+	require.NoError(t, bs.Commit())
+	require.NoError(t, bs.Close())
+
+	require.NoError(t, ex.Close(ctx))
+}
+
+func TestMultiEventTxnBlock(t *testing.T) {
+	t.Parallel()
+
+	// The test setup is executing a block that has the following transactions:
+	// - Txn 1, successful with two events: CREATE a table and INSERT a row.
+	// - Txn 2, failed with two events: CREATE a table and INSERT a row. INSERT has invalid SQL.
+	// - Txn 3, successful with one event: INSERT a row in the table created in Txn 1.
+	//
+	// Checks after block execution:
+	// 1) Txn 1 side-effects are seen in the db.
+	// 2) Txn 2 side-effects aren't seen in the db. In particular, the table created should have been rollbacked.
+	// 3) Txn 3 side-effects are seen in the db. Txn 2 failing should be isolated and have no effect in Txn 3 execution.
+	ctx := context.Background()
+	ex, dbURI := newExecutor(t, 0)
+
+	bs, err := ex.NewBlockScope(ctx, 0)
+	require.NoError(t, err)
+
+	// Txn 1
```

Comment:
I separated each Txn in a `{  ... } ` scope. You'll see the usual new mechanics of running tests reg creating txns and events.